js/libs/ui-shared/src/user-profile/LocaleSelector.tsx

```diff
@@ -20,9 +20,10 @@ export const LocaleSelector = ({
   supportedLocales,
 }: LocaleSelectorProps) => {
   const locales = supportedLocales.map((locale) => ({
-    key: locale,
-    value: localeToDisplayName(locale) || "",
-  }));
+      key: locale,
+      value: t("locale_" + locale, localeToDisplayName(locale) || locale),
+    }))
+    .sort((a, b) => a.value.localeCompare(b.value));
```

Comment:
Since this all depends on the locale configured by the user we have a special hook called [`useLocaleSort()`](https://github.com/keycloak/keycloak/blob/02e2ebf25829508945be6d5be12630935903d031/js/apps/admin-ui/src/utils/useLocaleSort.ts) that should be used instead.

Reply:
The hook is in the admin-ui, while this LocaleSelector is in the ui-shared. I assume it needs to be moved first, which seems to be out of reach for my JavaScript skills. Could this be part of a follow-up issue?

Reply:
This actually seems like something that makes sense to put in `ui-shared`, let me whip up a PR to do that for you.

Reply:
Hmm. On closer inspection, this will mean moving the `WhoAmI` context as well, which brings in a lot more work, let's save that for another time. What we can do is pass the current locale into `LocaleSelector` as a prop, that should be manageable.

Reply:
This has been resolved in 7802ed3f153c10e8d973a128ed48aa8bd2218bdb